bindings/java/src/main/java/org/apache/opendal/Capability.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal;
+
+import lombok.ToString;
+
+@ToString
+public class Capability {
```

Comment:
You can use `@Data` to avoid many boilerplate code.

Reply:
The field values are all read-only, so I don't think there's a need to add getter and setter methods, right?

Reply:
Then you may use `@RequiredArgsConstructor`

Reply:
And I think showing the declaration of the constructor method reduces the chances of errors in jni-rs calls.

Reply:
This makes sense. Since the order is significant..

Reply:
FWIW one day FFM is stable we may make something directly mapping instead of marshalling them manually..

FFM - https://openjdk.org/jeps/442